### PR TITLE
Improve error message for stale views

### PR DESCRIPTION
**Pull request by rschlussel**
Add information about why the view was considered stale to error message
about stale views

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: presto-main/src/test/java/com/facebook/presto/sql/analyzer/TestAnalyzer.java [hunk 1]
```diff
@@ -1643,14 +1645,34 @@ public void setup()
         inSetupTransaction(session -> metadata.createView(session, new QualifiedObjectName(TPCH_CATALOG, "s1", "v1"), viewData1, false));
 
         // stale view (different column type)
-        String viewData2 = JsonCodec.jsonCodec(ViewDefinition.class).toJson(
+        String viewData2a = JsonCodec.jsonCodec(ViewDefinition.class).toJson(
```
[the remark this answers is not in the document]
Comment (arhimondr) [reply presto-main/src/test/java/com/facebook/presto/sql/analyzer/TestAnalyzer.java]: It feels like `jsonCodec` has to be static important, as `JsonCodec.jsonCodec` is tautological .

File: presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java [hunk 2]
```diff
@@ -2035,23 +2033,37 @@ private Query parseView(String view, QualifiedObjectName name, Node node)
             }
         }
 
-        private boolean isViewStale(List<ViewDefinition.ViewColumn> columns, Collection<Field> fields)
+        private void checkViewNotStale(Table table, QualifiedObjectName name, List<ViewDefinition.ViewColumn> columns, Collection<Field> fields)
         {
             if (columns.size() != fields.size()) {
-                return true;
+                throw new SemanticException(VIEW_IS_STALE, table, "View '%s' is stale; it must be re-created:  Expected %s columns, but found %s", name, columns.size(), fields.size());
```
[the remark this answers is not in the document]
Comment (rschlussel) [reply presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java]: Seems useful, but some views have hundreds of columns, so not sure listing them out is a good idea.

File: presto-main/src/test/java/com/facebook/presto/sql/analyzer/TestAnalyzer.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (rschlussel) [reply presto-main/src/test/java/com/facebook/presto/sql/analyzer/TestAnalyzer.java]: +1.  I'll fix these.

**Comment by rschlussel** [comment]
> Curious about static imports. Also do we static import ImmutableList.of?

We don't static import ImmutableList.of().  I think the general rule is that if the method name is self explanatory we static import, and if it only makes sense with the class name attached we don't.

File: presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (aweisberg) [reply presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java]: I think it's probably not that useful if the issue is a stale view. Which columns don't matter you just recreate the view (or whatever the remediation is).

**Pull request closed without merging**